urlb3290c8e9e_craigonic_jasl_tgz-pJ8-DriverJ8/src/jasl/ui/data/Side.java

```diff
@@ -0,0 +1,379 @@
+
+		returnString.append("Side:\t" + toString() + "\n\n");
+
+		returnString.append("\tNationalities:\t");
+
+		ArrayList<Nationality.Nationalities> nationalities =
+			(Side.Sides.ALLIES == _side) ?
+			alliedNationalities() : axisNationalities();
+
+		for (Nationality.Nationalities nationality : nationalities)
+		{
+			returnString.append("\"" + nationality.toString() + "\" ");
+		}
+
+		returnString.append("\n\n");
+
+		for (Map.Entry<String,Player> player : _players.entrySet())
+		{
+			returnString.append(player.getValue().toText());
+		}
+
+		// Return the completed string to calling program.
+
+		return returnString.toString();
+	}
+
+	/**
+	 * Return the name of a side.
+	 *
+	 * @return a <CODE>String</CODE> specifying the side's name.
+	 */
+
+	public String toString()
+	{
+		return _side.toString();
+	}
+
+	/**
+	 * Return the names of all the players on a side.
+	 *
+	 * @return an <CODE>ArrayList</CODE> of the names.
+	 */
+
+	public ArrayList<String> playerList()
+	{
+		return new ArrayList<String>(_players.keySet());
+	}
+
+	/**
+	 * Provide access to the data, including units, for a player.
+	 *
+	 * @return the <CODE>Player</CODE> instance associated with the specified name. The
+	 * return value is null if a player with the name does not exist on the
+	 * side.
+	 */
+
+	public Player player(String name)
+	{
+		return (!_players.isEmpty()) ? _players.get(name) : null;
+	}
+
+	/**
+	 * Return the nationalities associated with the Allies.
+	 *
+	 * @return an <CODE>ArrayList</CODE> of the corresponding nationalities.
+	 */
+
+	public static ArrayList<Nationality.Nationalities> alliedNationalities()
+	{
+		ArrayList<Nationality.Nationalities> nationalities =
+			new ArrayList<Nationality.Nationalities>();
+
+		nationalities.add(Nationality.Nationalities.AMERICAN);
+		nationalities.add(Nationality.Nationalities.BRITISH);
+		nationalities.add(Nationality.Nationalities.RUSSIAN);
+		nationalities.add(Nationality.Nationalities.FRENCH);
+		nationalities.add(Nationality.Nationalities.PARTISAN);
+		nationalities.add(Nationality.Nationalities.ALLIED_MINOR);
+
+		return nationalities;
+	}
+
+	/**
+	 * Return the nationalities associated with the Axis.
+	 *
+	 * @return an <CODE>ArrayList</CODE> of the corresponding nationalities.
+	 */
+
+	public static ArrayList<Nationality.Nationalities> axisNationalities()
+	{
+		ArrayList<Nationality.Nationalities> nationalities =
+			new ArrayList<Nationality.Nationalities>();
+
+		nationalities.add(Nationality.Nationalities.GERMAN);
+		nationalities.add(Nationality.Nationalities.JAPANESE);
+		nationalities.add(Nationality.Nationalities.ITALIAN);
+		nationalities.add(Nationality.Nationalities.FINNISH);
+		nationalities.add(Nationality.Nationalities.AXIS_MINOR);
+
+		return nationalities;
+	}
+
+	// This method is deprecated. It should be removed after the Scenario
+	// and Stack classes are fully implemented.
+
+	public static ArrayList<String> unitList(Nationality.Nationalities nationality)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 245-252, lines 254-261, lines 263-270, lines 291-300, lines 302-311 into separate methods.